### DISCUSSION TITLE
Override macOS 15 feature with Rectangle

### DISCUSSION THREAD

**Discussion by honandreas** [rxhanson/Rectangle#1473] [General]
Hi! I've been using rectangle for a while and when the new macOS Sequoia launch and have the same feature, i think i still prefer rectangle over the macOS origin features.

At first there is a pop up that tell me to choose either i want to use Rectacle or macOS feature and i choose dismiss.

Then now i can't find that option again. I want to override macOS feature with Rectangle. Does anyone can help me?

Thanks!

**Comment by Kreachers** [comment]
After searching for an hour I finally found the setting.
Under "Desktop & Dock" scroll down to the "Windows" section there is a "Tile by dragging windows to screen edges" toggle. Turn that off and, at least I had to, toggle the snapping setting within Rectangle off and back on.

There is also a section there for disabling the spacing around the windows when using MacOS's snapping if someone does prefer MacOS's snapping but not the margin.

Now if only I could figure out how to stop windows from having rounded corners...

![Screenshot 2024-09-17 at 09 17 42](https://github.com/user-attachments/assets/12ab1afd-cb3c-4dab-b0f6-a4f91346b367)

**Reply by hummerbliss** [reply]
Thank you for the detailed description. Works for me ! I had the same issue (chose macos window management popup post upgrade) and then trying to get it working.

**Reply by honandreas** [reply]
Wow this solved the issue. Thanks!

**Comment by georgettica** [comment]
maybe also add a script in rectangle to allow this?

I found this URL for this
https://www.macrumors.com/2024/06/12/macos-sequoia-window-tiling/

and I think the `README.md` has support on this but it's not clear if I updated.

I was gonna update my config to have a script in case I move to a new machine, but this issue is good enough for now

**Reply by rxhanson** [reply]
> maybe also add a script in rectangle to allow this?

So that I'm clear, are you asking to have something exposed in Rectangle to disable the macOS tiling?

**Reply by georgettica** [reply]
yeah, something along these lines. have something in `Rectangle` to not only say this is an issue but also a path to follow to remediate

**Reply by rxhanson** [reply]
In the current version, when you check the box for drag to screen edge snapping in Rectangle's settings, it will check the macOS behavior and prompt if there's a conflict.

**Comment by johnalanwoods** [comment]
Hi folks - sorry to necro, 

Am I correct to say that disabling macOS's in built so that one can use Rectangle is essentially now disabling these two?

<img width="489" alt="Screenshot 2024-11-14 at 17 22 11" src="https://github.com/user-attachments/assets/0985c9fc-7183-446a-bacd-0e93944058a2">

Sorry, it's just not that clear how to ensure things are correctly setup, without this Rectangle and macOS seem to 'compete' - @rxhanson cc.

**Reply by Kreachers** [reply]
The top one is most important in my experience. Assuming you want Rectangle to do everything for you, you can just disable all 4 switches grouped together there. That's what I have done.
The "Hold Option key while dragging" could probably be left on since I don't think Rectangle has a feature like that.

**Reply by johnalanwoods** [reply]
Is it worth adding this to the readme?

**Reply by Kreachers** [reply]
Certainly wouldn't hurt, IMO.

**Reply by injust** [reply]
Personally, I disabled the first option, but left the 2nd one enabled and disabled the corresponding functionality in Rectangle instead.

I feel like the native "Drag windows to menu bar to fill screen" feature works better because it's compatible with "Drag windows to top of screen to enter Mission Control".

This way, you can drag to the top of the screen for Mission Control (I use this a lot to rearrange windows between spaces), but if you want to maximize the window instead, just stop on the menu bar before you hit the top of the screen.

**Reply by Aaron-Rumpler** [reply]
> if you want to maximize the window instead, just stop on the menu bar before you hit the top of the screen

@rxhanson Any reason Rectangle couldn't do the same?

**Reply by rxhanson** [reply]
> Is it worth adding this to the readme?

Sure, I'll add something on it.

Side note, when you check the box for drag to screen edge snapping in Rectangle's settings, it will check the macOS behavior and prompt if there's a conflict.

> if you want to maximize the window instead, just stop on the menu bar before you hit the top of the screen

The top margin for the snap area in Rectangle is 10, whereas the macOS default is the size of the menu bar. Perhaps it would make sense to adjust the default value in Rectangle to match. It can be manually adjusted via terminal command: 

```
defaults write com.knollsoft.Rectangle snapEdgeMarginTop -int 10
```

(https://github.com/rxhanson/Rectangle/blob/main/TerminalCommands.md#changing-the-margin-for-the-snap-areas)
